spec/lib/brexit_checker/question_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+
+describe BrexitChecker::Question do
+  let(:q1) { FactoryBot.build(:brexit_checker_question) }
+  let(:q2) { FactoryBot.build(:brexit_checker_question, criteria: [{ "all_of" => %w(a b) }]) }
+  let(:q3) { FactoryBot.build(:brexit_checker_question, criteria: [{ "all_of" => %w(c d) }]) }
+  let(:q4) { FactoryBot.build(:brexit_checker_question, criteria: %w(c)) }
+
+  let(:questions) { [q1, q2, q3, q4] }
+
+  describe '#resolve' do
+    before do
+      allow(BrexitChecker::Question).to receive(:load_all) { questions }
+    end
+
+    let(:subject) {
+      BrexitChecker::Question.resolve(
+        previous_question: previous_question, criteria_keys: criteria_keys
+      )
+    }
+
+    let(:criteria_keys) { [] }
+    let(:previous_question) { nil }
+
+    context 'the page on the form is 0' do
+      let(:previous_question) { 0 }
+      it 'returns question one ' do
+        expect(subject).to eq(q1)
+      end
+    end
+
+    context 'the page on the form is 1 and the criteria_keys match question_two' do
+      let(:previous_question) { 1 }
+      let(:criteria_keys) { %w[c d] }
+      it 'returns question three' do
+        expect(subject).to eq(q3)
+      end
+    end
+
+    context 'the page on the form is 1 and the criteria_keys match question_three' do
```

Comment:
Minor: could we rephrase this to describe why this is a special case? The setup is pretty self-evident below, so we have the space to write something more descriptive.